joomla/joomla-cms
# Add ability to switch off the animation of showons

### Summary of Changes
To conditionally hide fields or content based on the values of other fields, you can add a `showon=""` attribute to fields in xml files.

However, this will always use a slideUp/slideDown animation.
There are cases where you do not want to animate the showing/hiding of your content or fields.

This patch adds the ability to switch off the animation by adding a class`no-animation` or `no-animate` to the field.
You can also add the class to the surrounding showon container, which can be helpful when creating your own via php.

### Testing Instructions
For instance, you can test with the Joomla core configuration:
```
		<field
			name="page_heading"
			type="text"
			label="COM_MENUS_ITEM_FIELD_PAGE_HEADING_LABEL"
			description="COM_MENUS_ITEM_FIELD_PAGE_HEADING_DESC"
			default=""
			showon="show_page_heading:1"
		/>
```
And change that to:
```
		<field
			name="page_heading"
			type="text"
			label="COM_MENUS_ITEM_FIELD_PAGE_HEADING_LABEL"
			description="COM_MENUS_ITEM_FIELD_PAGE_HEADING_DESC"
			default=""
			showon="show_page_heading:1"
			class="no-animation"
		/>
```
Now go to: `administrator/index.php?option=com_config&view=component&component=com_menus` and toggle the `Show Page Heading` setting.
Before the patch it should have the slide animation. After, it should not animate.

### Documentation Changes Required
https://docs.joomla.org/Form_field
Optionally add a note somewhere that you can switch off the animation effect via the classname.
